### PR TITLE
Add configuration and wrapper explainers

### PR DESCRIPTION
**Pull request by keturiosakys**
This PR moves the expanded documentation about the TypeScript library to the guides section here. It also updates the sidebar default expansion level.

- set default menu collapse to 1
- add the right side bar config
- add configuration docs
- add wrapper docs

### CODE REVIEW

File: src/pages/typescript/using-wrappers-and-decorators.mdx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+Use the Autometrics wrappers and decorators to instrument the functions you want
+to track. Usually it makes sense to wrap functions that are doing some business
+logic, making network calls, or are otherwise critical to the service (e.g.:
+request handlers or database calls).
+
+### Wrapping plain-old functions
+
+Wrappers are simple functions that wrap the original function declaration and
+instrument it with metrics. They allow the language service plugin to add
```
Comment (emschwartz) [inline src/pages/typescript/using-wrappers-and-decorators.mdx:15]:
```suggestion
instrument it with metrics. They also enable the language service plugin to add
```

Does it also enable the VS Code extension to do this?

File: src/pages/typescript/using-wrappers-and-decorators.mdx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+additional information in the type docs.
+
+Use function wrappers to wrap request handlers, database calls, or other
+pieces of important business logic that you want to measure.
+
+<Callout type="info">
+    Wrapped functions must be named. Autometrics will, not affect the wrapped
```
Comment (emschwartz) [inline src/pages/typescript/using-wrappers-and-decorators.mdx:22]: "Autometrics will, not..." this doesn't make sense

Also might be worth saying something like "named (as opposed to anonymous arrow functions)"

File: src/pages/typescript/using-wrappers-and-decorators.mdx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  // ...
+});
+
+createUserWithMetrics();
+```
+
+### Decorating class methods
+
+When using a decorator for a class method, it is wrapped in additional code that
+instruments the method with OpenTelemetry metrics.
```
Comment (emschwartz) [inline src/pages/typescript/using-wrappers-and-decorators.mdx:41]:
```suggestion
instruments the method with metrics.
```

**Review by oscarvz — APPROVED** [review]
It needs some small copy changes, but other than that looking good! 🗒️

File: src/pages/typescript/using-wrappers-and-decorators.mdx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+```typescript
+init({ pushGateway: "<link_to_gateway>" });
+```
+
+### Use Autometrics wrapper with options
+
+The same wrapper functions can be used in browser environments. Note that
+bundlers often change the names of functions and modules in production, which
+can impact the library.
```
[the remark this answers is not in the document]
Comment (keturiosakys) [reply src/pages/typescript/using-wrappers-and-decorators.mdx:96]: 👍

**Pull request merged**